test/functional/sessions_controller_test.rb

```diff
@@ -201,6 +201,89 @@ class SessionsControllerTest < ActionController::TestCase
 
       should respond_with :unauthorized
     end
+
+    context "when user has mfa enabled" do
+      setup do
+        @user = create(:user, :mfa_enabled)
+        post(
+          :create,
+          params: { session: { who: @user.handle, password: @user.password } }
+        )
+      end
+
+      should respond_with :ok
+
+      should "not set webauthn_authentication" do
+        assert_nil session[:webauthn_authentication]
+      end
+
+      should "set mfa_user" do
+        assert_equal @user.id, session[:mfa_user]
+      end
+
+      should "have mfa forms and not webauthn credentials form" do
+        assert page.has_content?("multi-factor authentication")
+        assert page.has_field?("OTP or recovery code")
+        assert page.has_button?("Sign in")
+      end
+    end
+
+    context "when user has webauthn credentials" do
+      setup do
+        @user = create(:user)
+        create(:webauthn_credential, user: @user)
+        post(
+          :create,
+          params: { session: { who: @user.handle, password: @user.password } }
+        )
+      end
+
+      should respond_with :ok
+
+      should "set webauthn authentication" do
+        assert_equal @user.id, session[:webauthn_authentication]["user"]
+        assert_not_nil session[:webauthn_authentication]["challenge"]
+      end
+
+      should "not set mfa_user" do
+        assert_nil session[:mfa_user]
+      end
+
+      should "not have mfa forms and have webauthn credentials form" do
+        assert page.has_content?("Multi-factor authentication")
```

Comment:
I see different capitalisation of "Multi-factor authentication" vs "multi-factor authentication" at L255 and L282.

Reply:
https://github.com/rubygems/rubygems.org/search?q=Multi-factor+authentication it seems dependent on position in sentence